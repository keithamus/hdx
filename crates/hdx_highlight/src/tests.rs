use super::test_helpers::*;
use super::*;

#[test]
fn test_basic() {
	assert_highlight!(
		"basic",
		r#"body {
			appearance: initial;
			color: blue;
			will-ever-exist: foo;
		}"#,
	);
}

#[test]
fn test_tags() {
	assert_highlight!("tags", r#"body, dialog, madeup, marquee, portal, custom-element {}"#);
}

#[test]
fn test_nesting() {
	assert_highlight!("nesting", r#"body{main{dialog{input{}}}}"#);
}

#[test]
fn test_pseudo_classes() {
	assert_highlight!("pseudo_classes", r#"body:focus,dialog:modal{}"#);
}
