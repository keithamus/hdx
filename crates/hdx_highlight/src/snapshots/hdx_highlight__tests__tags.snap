---
source: crates/hdx_highlight/src/tests.rs
expression: actual
---
<!DOCTYPE html>
<head>
	<style>
		:root { background: #22272E; color: hotpink }
		.tag { color: #8ddb8c }
		.punctuation { color: #d1d7e0 }
		.property { color: #6cb6ff }
		.pseudo-class { color: #6cb6ff }

		.unknown { color: grey }
		.deprecated { text-decoration: line-through }
		.experimental { text-decoration: wavy underline #bf4b8a 0.5px }
	</style>
</head>
<body>
	<pre>
		<code>
		<span class="Tag">body</span>,<span class="Tag">dialog</span>,<span class="Tag unknown">madeup</span>,<span class="Tag deprecated">marquee</span>,<span class="Tag experimental">portal</span>,<span class="Tag custom">custom-element</span><span class="Punctuation">{</span><span class="Punctuation">}</span>
					</code>
				</pre>
			</body>
